pnet_sys/src/unix.rs

```diff
@@ -28,7 +28,46 @@ pub mod public {
     pub const IP_HDRINCL: libc::c_int = libc::IP_HDRINCL;
 
     pub const IFF_LOOPBACK: libc::c_int = libc::IFF_LOOPBACK;
+    
+    #[cfg(target_os = "macos")]
+    mod device_flags {
+        use super::libc;
+        // XNU
+        /// https://github.com/aosm/xnu/blob/master/bsd/net/if.h#L140-L156
+        pub const IFF_UP: libc::c_int          = 0x1;       //  interface is up
+        pub const IFF_BROADCAST: libc::c_int   = 0x2;       //  broadcast address valid
+        pub const IFF_DEBUG: libc::c_int       = 0x4;       //  turn on debugging
```

Comment:
There are flags added here that is never unused by this PR itself. Such as this one, `IFF_DEBUG`, and `IFF_NOTRAILERS`. Why do we need to expose these flags if we don't use them. What can a user of this library do with them?

Reply:
I think this is fine - it's better to have the complete set for these things.

It's a huge hassle trying to add just one at a later point - you have to find the right header files for the right platform, then find the right constant, figure out the right way to translate it (there's usually more than one way, and which is best is never obvious), then figure out where it needs to go on the Rust side... A lot more effort than just doing it all at once. Even if some of the values turn out to be wrong, it's easier to fix them up than add them from scratch.

Reply:
@faern in my case, i just want to know net interface device type (tun or tap).